SubscriptionService/SubscriptionService.DataAccess/Repositories/Implementations/UnitOfWork.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using SubscriptionService.DataAccess.Data;
+using SubscriptionService.DataAccess.Repositories.Interfaces;
+
+namespace SubscriptionService.DataAccess.Repositories.Implementations
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private readonly SubscriptionDbContext _dbContext;
+        private readonly IServiceProvider _serviceProvider;
+        private ISubscriptionRepository? _subscriptions = null;
+        private ITariffPlanRepository? _tariffPlans = null;
+
+        public ISubscriptionRepository Subscriptions => _subscriptions ??= _serviceProvider.GetService<ISubscriptionRepository>()!;
+        public ITariffPlanRepository TariffPlans => _tariffPlans ??= _serviceProvider.GetService<ITariffPlanRepository>()!;
```

Comment:
for better code readability use a traditional properties with a getter and a setter instead of the expression-bodied member